### PR TITLE
Internal Java Solution for Measuring Traffic

### PR DESCRIPTION
**Pull request by brad182**

### CODE REVIEW

File: solutions/usaco-917.mdx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class MeasuringTraffic {
+	static final int infinity = 999999999;
```
Comment (nathangong) [inline solutions/usaco-917.mdx:22]: Integer.MAX_VALUE is better for this
```suggestion
	static final int infinity = Integer.MAX_VALUE;
```

Comment (ryanchou-dev) [inline solutions/usaco-917.mdx:22]: would something like integer.max_value be better for this?

File: solutions/usaco-917.mdx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		int[] start = new int[numMiles];
+		int[] end = new int[numMiles];
+
+		for (int x = 0; x < numMiles; x++) {
+			segmentType[x] = io.next();
+			start[x] = io.nextInt();
+			end[x] = io.nextInt();
+		}
+
+		// initially, set the range extremely large
```
Comment (ryanchou-dev) [inline solutions/usaco-917.mdx:37]:
```suggestion
		// initially, set an extremely large range 
```

File: solutions/usaco-917.mdx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+				low -= end[i];
+				high -= start[i];
+				low = Math.max(0, low);
+			}
+		}
+
+		io.println(low + " " + high);
+		io.close();
+	}
+	//CodeSnip{Kattio}
```
Comment (nathangong) [inline solutions/usaco-917.mdx:85]:
```suggestion
	// CodeSnip{Kattio}
```

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
code is great, just some minor changes :)

File: solutions/usaco-917.mdx (hunk 1 above)
Comment (nathangong) [reply solutions/usaco-917.mdx:22]: great minds think alike 😆

Comment (SansPapyrus683) [reply solutions/usaco-917.mdx:22]: or better yet do away with the constant entirely and just use `Integer.MAX_VALUE`

File: solutions/usaco-917.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/usaco-917.mdx:85]: don't think this is needed?

File: solutions/usaco-917.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-917.mdx:22]: wait nvm
overflow is a possibility

File: solutions/usaco-917.mdx (hunk 3 above)
Comment (nathangong) [reply solutions/usaco-917.mdx:85]: Just for consistency with the rest of the comments

File: solutions/usaco-917.mdx (hunk 1 above)
Comment (nathangong) [reply solutions/usaco-917.mdx:22]: that works too, it won't overflow because the bounds will be updated by the time you add/subtract stuff

Comment (SansPapyrus683) [reply solutions/usaco-917.mdx:22]: image
no i think it will

Comment (nathangong) [reply solutions/usaco-917.mdx:22]: nah, when you do Math.max, low will instantly update to whatever start[i[ is. Same with high.

Comment (SansPapyrus683) [reply solutions/usaco-917.mdx:22]: bold of you to assume that the max will be executed first

Comment (SansPapyrus683) [reply solutions/usaco-917.mdx:22]: wait also @brad182 can you convert these to else ifs

Comment (nathangong) [reply solutions/usaco-917.mdx:22]: whoops you're right

File: solutions/usaco-917.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/usaco-917.mdx:85]: but this isn't a comment it's a codesnip

Comment (brad182) [reply solutions/usaco-917.mdx:85]: i never do this, and in the real thing it just shows up as a dropdown and the comment doesn't even show

File: solutions/usaco-917.mdx (hunk 1 above)
Comment (brad182) [reply solutions/usaco-917.mdx:22]: yep he's right and i tried it too. the second if executes first in the sample input

Comment (SansPapyrus683) [reply solutions/usaco-917.mdx:22]: > wait also @brad182 can you convert these to else ifs

also add a comment explaining why integer.max_value wouldn't work

**Review by ryanchou-dev — APPROVED** [review]
think you still gotta address this: [discussion comment](https://github.com/cpinitiative/usaco-guide/pull/1987#discussion_r676159709). but the rest is good!

**Review by nathangong — APPROVED** [review]

**Review by ar492 — APPROVED** [review]
👍

**Comment by maggie-j-liu** [comment]
#1856 exists :/

**Review by nwatx — APPROVED** [review]

**Review by SansPapyrus683 — APPROVED** [review]
i give up

**Review by maggie-j-liu — APPROVED** [review]
I think it looks good, but there's already a pr for this problem (but seems to be stale? @4Gleneyre ?). i think it would be fine to merge this and have the other pr replace when it's finished, but does anyone else have an opinion?

**Comment by 4Gleneyre** [comment]
I think this looks good as well.  I think you can just merge this and disregard my PR.

**Comment by brad182** [comment]
> I think it looks good, but there's already a pr for this problem (but seems to be stale? @4Gleneyre ?). i think it would be fine to merge this and have the other pr replace when it's finished, but does anyone else have an opinion?

ill search the pull requests list next time, especially on bronze ones that a lot of people can do. also maybe there should be a two week inactive policy or something

**Comment by maggie-j-liu** [comment]
> I think this looks good as well. I think you can just merge this and disregard my PR.

but i thought we wanted the code and explanations that match the videos to be on the guide? @thecodingwizard ?

**Review by dongliu0426 — APPROVED** [review]

**Review by ar492 — APPROVED** [review]

**Pull request merged**